eslint.config.mjs

```diff
@@ -157,6 +160,93 @@ export default config(
       "no-useless-concat": "warn",
       "no-useless-rename": "error",
 
+      // === JSDoc =============================================================
+      // (We are assuming that the config is extended by JSDoc's: recommended
+      // extension)
+
+      // Correctness
+      // TODO: Requires manual fixes, enable in a separate PR.
+      "jsdoc/check-param-names": "off",
+      "jsdoc/check-tag-names": [
+        "error",
+        {
+          // Non-standard JSDoc tags we use to generate documentation; see
+          // docs/src/lib/transform.ts.
+          definedTags: [
+            "signature",
+            "exampleRaw",
+            "dataFirst",
+            "dataLast",
+            "indexed",
+            "pipeable",
+            "strict",
+            "category",
+          ],
+        },
+      ],
+      "jsdoc/no-bad-blocks": "error",
+      "jsdoc/no-blank-blocks": "error",
+      "jsdoc/require-asterisk-prefix": "error",
+
+      // Completeness
+      // TODO: Requires manual fixes, enable in a separate PR.
+      "jsdoc/require-jsdoc": [
+        "off",
+        {
+          enableFixer: false,
+          // We only require JSDoc for top-level function exports. Assuming
+          // that each function has overrides, we only require JSDocs for the
+          // overrides (which are TSDeclareFunction) and not the implementation
+          // (which is FunctionDeclaration).
+          require: { FunctionDeclaration: false },
+          contexts: ["Program > ExportNamedDeclaration > TSDeclareFunction"],
+        },
+      ],
+      "jsdoc/require-description": ["error", { exemptedBy: ["deprecated"] }],
+      "jsdoc/require-example": [
+        "warn",
+        { enableFixer: false, exemptedBy: ["deprecated", "exampleRaw"] },
+      ],
+      // TODO: Requires manual fixes, enable in a separate PR.
+      "jsdoc/require-param": "off",
+      // TODO: Requires manual fixes, enable in a separate PR.
+      "jsdoc/require-returns": "off",
+
+      // Style
+      "jsdoc/no-multi-asterisks": ["warn", { allowWhitespace: true }],
+      "jsdoc/require-description-complete-sentence": [
+        "warn",
+        { abbreviations: ["etc.", "e.g.", "i.e."] },
+      ],
+      "jsdoc/require-hyphen-before-param-description": [
+        "error",
+        "always",
+        { tags: { "*": "never" } },
+      ],
```

Comment:
these two are the culprits behind most of the changes in this pr. we can turn them off

Reply:
I think this is fine... I'm assuming this PR only contains auto-fixers and no manual fixes, right?

Reply:
yep!